dojo/importers/importer/importer.py

```diff
@@ -91,6 +91,10 @@ def process_parsed_findings(self, test, parsed_findings, scan_type, user, active
                 item.active = active
             if item.verified:
                 item.verified = verified
+            # Only set date value of the test if they were NOT set by the parser
+            # scan_date is set to both target_start and target_end in the creat_test function
+            if item.date:
```

Comment:
Unfortunately this doesn't work:
- When the parser sets a `date` explicitly, `item.date` is true and it will be overwritten with `test.target_start`. This is not what we want.
- When the parser doesn't set a `date`, the model sets `current_date` as a default

So basically the `date` is set here, but you don't know why. I thought about a logic comparing the dates, but couldn't find one that works, as long as there is the default in the model.

Reply:
I did not think the date would actually be present until the finding was saved (and the default date would be applied)

Do you think it would be good to set to
```
if **not** item.date:
    item.date = test.target_date
```

Reply:
I tried that, it doesn't work as well, because the model has set the date to current date as a default already. It would work when we remove the default, but then we might get other side effects.

Reply:
I've got a quick idea that is a bit more fleshed out. Will commit it and ping you

Reply:
I am afraid we need more test cases:
- Parser sets date explicitly
- Import sets date different from today

Reply:
The first branch works in all but one cases: When the date for the import is set to something different than today but the parser has set the date explicitely to today. Then the parser date would be overriden. But I think this shouldn't happen in reality and we can ignore that.

I am honestly not sure about the else branch, do we want to override the date the parser has set, when the user set a date for the import? When I think about it while writing, I think it makes sense. Would like to hear the opinion of other maintainers.

Do we need to the same for re-imports?

Reply:
The import `scan_date` should overwrite the date the parser recorded. It is the user taking it into their hands to change it, so the findings should record as such.

I am not sure about re-import as I feel it's primary use is for automated imports as well as keeping an up to date view of what's vulnerable.

The main motivation for this PR is because when importing older reports, the date of the test was often far in the future than the findings themselves. I do no think many would be re-importing older scan reports.

Reply:
I am really open about it. @valentijnscholten and @damiencarol, you know import and reimport the best, what do you think?

Reply:
I think the idea of this PR is good, but we need some test cases (haven't checked if there are any). We want to make sure it fully works as expected. Otherwise it gets confusing, for example what we saw with the `active/verifed` being overwritten or not by the parsers or importers. I think during reimport the date of the finding was already set during the initial import, so probably no need to add logic there.

I haven't checked the code, but could it work to remove the default from the default and only set a default in `save()` as safeguard? That way the `if not item.date` could work.

Reply:
There is a very basic test (just showing that the date at import time does overwrite the default date), but the scan that is used in tests (zap) does not set the date on the report level,  so I would need to test with another scanner (I was using Acunetix from the unit tests when I wrote this)

Sounds like leaving re-import alone is the move though.

Reply:
> For example, here are some scenarios of how this PR affects the date of a finding:
>
>    * The parser does not set the date, and scan date is not specified at import
>    Finding date will be the default value of scan date (time of import)
>    * The parser sets the date, and the scan date is not specified at import
>    Finding date will be whatever the parser sets
>    * The parser does not set the date, and the scan date is specified at import
>    Finding date will be whatever the user specified for scan date
>    * The parser sets the date, and the scan date is specified at import
>    Finding date will be whatever the user specified for scan date

I fully agree.

Specifically this comment:
> The import scan_date should overwrite the date the parser recorded. It is the user taking it into their hands to change it, so the findings should record as such.

It's why I think this PR is good. More control to the user and a way to override the data of the parser *OR* let the parser do the job and use the data as-is.

Also:
> I think the idea of this PR is good, but we need some test cases (haven't checked if there are any). We want to make sure it fully works as expected. Otherwise it gets confusing, for example what we saw with the active/verifed being overwritten or not by the parsers or importers. I think during reimport the date of the finding was already set during the initial import, so probably no need to add logic there.

I think @valentijnscholten is right. To make this PR works we need to be good on the unit tests part. We must have the 4 cases covered.

> I haven't checked the code, but could it work to remove the default from the field and only set a default in save() as safeguard? That way the if not item.date could work.

Same here, I haven't checked the code but I expect that the `save()` logic is changed to allow the logic described in the PR description.

Also, @Maffooch it could be good to make it clear in the documentation how we handle the date. Could you add a part in the documentation that clearly state the 1/2/3/4 cases? Thanks.